editor/src/components/canvas/canvas-strategies/post-action-options/post-action-paste.ts

```diff
@@ -240,7 +298,188 @@ export const PasteWithPropsReplacedPostActionChoice = (
+  ) {
+    return null
+  }
+  return {
+    name: 'Paste here with variables replaced',
+    id: PasteHereWithPropsReplacedPostActionChoiceId,
+    run: (editor, derived, builtInDependencies) => {
+      if (
+        editor.internalClipboard.elements.length !== 1 ||
+        editor.internalClipboard.elements[0].copyDataWithPropsReplaced == null
+      ) {
+        return []
+      }
+      const elementToPaste = editor.internalClipboard.elements[0].copyDataWithPropsReplaced.elements
+      const targetParent = getTargetParentForPasteHere(editor, derived, elementToPaste)
+      if (isLeft(targetParent)) {
+        return [
+          showToastCommand(
+            'Paste element cannot find parent',
+            'WARNING',
+            'paste-elements-cannot-find-parent',
+          ),
+        ]
+      }
+
+      const originalMetadata =
+        editor.internalClipboard.elements[0].copyDataWithPropsPreserved
+          .targetOriginalContextMetadata
+      const originalPathTree =
+        editor.internalClipboard.elements[0].targetOriginalContextElementPathTrees
+
+      return pasteChoiceCommon(
+        targetParent.value,
+        {
+          builtInDependencies: builtInDependencies,
+          nodeModules: editor.nodeModules.files,
+          openFile: editor.canvas.openFile?.filename ?? null,
+          pasteTargetsToIgnore: [],
+          projectContents: editor.projectContents,
+          startingMetadata: editor.jsxMetadata,
+          startingElementPathTrees: editor.elementPathTree,
+          startingAllElementProps: editor.allElementProps,
+        },
+        {
+          selectedViews: editor.selectedViews,
+          elementPasteWithMetadata: {
+            elements: elementToPaste,
+            targetOriginalContextMetadata: originalMetadata,
+          },
+          targetOriginalPathTrees: originalPathTree,
+          canvasViewportCenter: zeroCanvasPoint,
+          reparentStrategy: 'REPARENT_AS_ABSOLUTE',
+          insertionPosition: data.position,
+        },
+      )
+    },
+  }
+}
+
+function getTargetParentForPasteHere(
+  editor: EditorState,
+  derived: DerivedState,
+  elementToPaste: Array<ElementPaste>,
+): Either<string, ReparentTargetForPaste> {
+  const originalMetadata =
+    editor.internalClipboard.elements[0].copyDataWithPropsPreserved.targetOriginalContextMetadata
+  const originalPathTree =
+    editor.internalClipboard.elements[0].targetOriginalContextElementPathTrees
+
+  const target = getTargetParentForPaste(
+    editor.projectContents,
+    editor.selectedViews,
+    editor.nodeModules.files,
+    editor.canvas.openFile?.filename ?? null,
+    editor.jsxMetadata,
+    editor.pasteTargetsToIgnore,
+    {
+      elementPaste: elementToPaste,
+      originalContextMetadata: originalMetadata,
+      originalContextElementPathTrees: originalPathTree,
+    },
+    editor.elementPathTree,
+  )
+  if (isLeft(target)) {
+    return target
+  }
```

Comment:
If there's no target we should probably fall back to trying to paste onto the storyboard here

Reply:
good idea, fixed